### PR TITLE
Update DiscoverGranules to get provider_path from config, not config.collection

### PR DESCRIPTION
**Pull request by yjpa7145**
The `DiscoverGranules` task previously expected its `provider_path` to be specified in `event.config.collection.provider_path`. It is now expected in `event.config.provider_path`. This is a step toward [CUMULUS-1969](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1969): "Set provider_path on the rule instead of the collection."

### CODE REVIEW

File: example/cumulus-tf/browse_example.tf [hunk 1]
```diff
@@ -19,6 +19,7 @@ module "discover_granules_browse_example_workflow" {
           "event.$": "$",
           "task_config": {
             "provider": "{$.meta.provider}",
+            "provider_path": "{$.meta.collection.provider_path}",
```
Comment (markdboyd) [inline example/cumulus-tf/browse_example.tf:22]: Do we want to also update the ingest browse data cookbook to specify that the rule should have the provider path?

https://nasa.github.io/cumulus/docs/data-cookbooks/browse-generation

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### BREAKING CHANGES
 
+- **CUMULUS-1969**
```
[the remark this answers is not in the document]
Comment (yjpa7145) [reply CHANGELOG.md]: Done: caa432feb

File: example/cumulus-tf/browse_example.tf (hunk 1 above)
Comment (yjpa7145) [reply example/cumulus-tf/browse_example.tf:22]: Done: caa432feb

Comment (markdboyd) [reply example/cumulus-tf/browse_example.tf:22]: I'm wondering if it should also be updated here: https://nasa.github.io/cumulus/docs/data-cookbooks/run-tasks-in-lambda-or-docker

The purpose of that doc isn't necessarily to show a "working" discover granules workflow, but people might easily copy/paste from it and be confused when it doesn't work

Comment (yjpa7145) [reply example/cumulus-tf/browse_example.tf:22]: Fixed here: 1b5706565

**Pull request merged**